R/reportSE.R

```diff
@@ -61,13 +61,19 @@ reportSE <- function(gdx, regionSubsetList = NULL, t = c(seq(2005, 2060, 5), seq
 
   ## parameter
   pm_prodCouple    <- readGDX(gdx, c("pm_prodCouple", "p_prodCouple", "p_dataoc"), restore_zeros = FALSE, format = "first_found")
+
   pm_prodCouple[is.na(pm_prodCouple)] <- 0
+  teprodCoupleSeel <- getNames(mselect(pm_prodCouple, all_enty2 = "seel"), dim = 3)
+  if (!"ccsinje" %in% teprodCoupleSeel) {
+    pm_prodCoupleEmi <- readGDX(gdx, c("pm_prodCoupleEmi"), restore_zeros = FALSE, format = "first_found")
+    pm_prodCoupleEmi[is.na(pm_prodCoupleEmi)] <- 0
```

Comment:
looks a bit strange that adding this set with value =0 is equal to the old version where "ccsinje" was included... Was ist always 0? Or should another parameter be used now?

Reply:
I also don't think this will work.

As far as I understand, you just need to test if you have pm_prodCoupleEmi in the model, in positive case (new formulation) you just load its values from the gdx, in negative case (old formulation) you simply assume its values equal to pm_prodCouple[,,ccinje], no?

```
pm_prodCoupleEmi <- readGDX(gdx, c("pm_prodCoupleEmi"), restore_zeros = FALSE, format = "first_found")
if (!(is.null(pm_prodCoupleEmi))){
pm_prodCouple[,ccinje] <- pm_prodCoupleEmi 
}

This way you wouldn't need to change anything else in the code, no?

Reply:
@LaviniaBaumstark sorry, I don't really understand your comment. What do you mean by "adding this set with value =0"?

Reply:
@Renato-Rodrigues You are right! Less change in the code is required if `pm_prodCoupleEmi` is just added back to `pm_prodCoupleEmi` in case it is separated. I commited a version of your code draft which uses `mbind` to achieve that. By that I don't even have to check with `is.null()` as `mbind(x, NULL)` is the same as `x`.